src/providers/authenticationProvider.ts

```diff
@@ -156,74 +171,46 @@ export class TerraformCloudAuthenticationProvider implements vscode.Authenticati
   // This function is called when the end user signs out of the account.
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   async removeSession(_sessionId: string): Promise<void> {
-    this.logger.info('Detecting current logged in session');
-    const session = (await this.getSessions())[0];
-
-    this.logger.info('Removing current session');
-    await this.secretStorage.delete(TerraformCloudAuthenticationProvider.secretKey);
-
-    this._onDidChangeSessions.fire({ added: [], removed: [session], changed: [] });
-  }
-
-  private async ensureInitialized(): Promise<void> {
-    if (this.initializedDisposable !== undefined) {
+    const session = await this.sessionPromise;
+    if (!session) {
       return;
     }
 
-    await this.cacheTokenFromStorage();
+    this.logger.info('Removing current session');
+    await this.sessionHandler.delete();
 
-    this.initializedDisposable = vscode.Disposable.from(
-      // This onDidChange event happens when the secret storage changes in _any window_ since
-      // secrets are shared across all open windows.
-      this.secretStorage.onDidChange((e) => {
-        if (e.key === TerraformCloudAuthenticationProvider.secretKey) {
-          this.logger.info('Authentication secret storage change');
-          void this.checkForUpdates();
-        }
-      }),
-      // This fires when the user initiates a "silent" auth flow via the Accounts menu.
-      vscode.authentication.onDidChangeSessions((e) => {
-        if (e.provider.id === TerraformCloudAuthenticationProvider.providerID) {
-          this.logger.info('Authentication provider change');
-          void this.checkForUpdates();
-        }
-      }),
```

Comment:
We're really sure we don't need this? What is the harm in keeping this?

Reply:
I looked at the different authentication examples and found that only the [`authenticationprovider-sample`](https://github.com/microsoft/vscode-extension-samples/tree/0db7faa03cb9fffb9c7d99f934e3c5591207c064/authenticationprovider-sample) has both change handlers in place. Unfortounaly I wasn't able to find a git history or any other information why both are introduced.
[github-authentication](https://github.com/microsoft/vscode/blob/e7e97f8ea8519885e13dac9df6eae7c747d531b1/extensions/github-authentication/src/github.ts#L84) and [microsoft-authentication](https://github.com/microsoft/vscode/blob/e7e97f8ea8519885e13dac9df6eae7c747d531b1/extensions/microsoft-authentication/src/betterSecretStorage.ts#L35) rely only on `secrets.onDidChange`.

Keeping both handlers will fire all events twice whenever there is a change, which seems unnecessary.

Reply:
👍🏼